### PR TITLE
refactor(ModuleManager): factor initiliaztion code to multiple methods

### PR DESCRIPTION
**Pull request by keturn**
Refactoring in anticipation of changes for gestalt v7.

I think these are all strict refactors, minimal testing required.

Open to feedback about where these methods should be ordered in the class file.

### Outstanding before merging

- [x] Merge parent #4539, which I based this on because that moved the ModuleManager class file. The diff on this will be clearer once that is merged.

### CODE REVIEW

File: engine/src/main/java/org/terasology/engine/module/ModuleManager.java [hunk 1]
```diff
@@ -79,16 +79,20 @@ public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspa
 
         loadModulesFromApplicationPath(pathManager);
 
+        ensureModulesDependOnEngine(engineModule);
```
Comment (keturn) [inline engine/src/main/java/org/terasology/engine/module/ModuleManager.java:82]: I wondered why this is important, and found https://github.com/MovingBlocks/Terasology/issues/1450

It's not a worry that the `engine` module wouldn't be loaded without it. It's about ordering: some things run in an order derived from the dependency tree, and we want to make sure engine is at the root of it.

Comment (skaldarnar) [reply engine/src/main/java/org/terasology/engine/module/ModuleManager.java:82]:
```suggestion
        // I wondered why this is important, and found #1450
        // It's not a worry that the engine module wouldn't be loaded without it. 
        // It's about ordering: some things run in an order derived from the dependency 
        // tree, and we want to make sure engine is at the root of it.
        ensureModulesDependOnEngine(engineModule);
```

File: engine/src/main/java/org/terasology/engine/module/ModuleManager.java [hunk 2]
```diff
@@ -125,6 +134,18 @@ public ModuleManager(Config config, List<Class<?>> classesOnClasspathsToAddToEng
         this(config.getNetwork().getMasterServer(), classesOnClasspathsToAddToEngine);
     }
 
+    private ModuleMetadataJsonAdapter newMetadataReader() {
+        final ModuleMetadataJsonAdapter metadataJsonAdapter;
+        metadataJsonAdapter = new ModuleMetadataJsonAdapter();
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/engine/module/ModuleManager.java:139]: Why not do this in one line?
```suggestion
        final ModuleMetadataJsonAdapter metadataJsonAdapter = new ModuleMetadataJsonAdapter();
```

File: engine/src/main/java/org/terasology/engine/module/ModuleManager.java [hunk 3]
```diff
@@ -68,13 +68,40 @@ public ModuleManager(String masterServerAddress) {
...
+        loadModulesFromApplicationPath(pathManager);
+
+        ensureModulesDependOnEngine(engineModule);
+
+        setupSandbox();
+        loadEnvironment(Sets.newHashSet(engineModule), true);
+        installManager = new ModuleInstallManager(this, masterServerAddress);
+    }
+
+    private void ensureModulesDependOnEngine(Module engineModule) {
```
Comment (skaldarnar) [inline engine/src/main/java/org/terasology/engine/module/ModuleManager.java:89]:
```suggestion
    /**
     * I wondered why this is important, and found MovingBlocks/Terasology#1450.
     * It's not a worry that the engine module wouldn't be loaded without it. 
     * It's about ordering: some things run in an order derived from the dependency 
     * tree, and we want to make sure engine is at the root of it.
     */   
    private void ensureModulesDependOnEngine(Module engineModule) {
```

File: engine/src/main/java/org/terasology/engine/module/ModuleManager.java (hunk 1 above)
Comment (skaldarnar) [reply engine/src/main/java/org/terasology/engine/module/ModuleManager.java:82]: Added this as comment to `ensureModulesDependOnEngine` for later reference.

**Review by skaldarnar — CHANGES_REQUESTED** [review]

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**